### PR TITLE
Add google-java-format

### PR DESCRIPTION
**Pull request by mxr**
Copied the same structure as `ktfmt`

### CODE REVIEW

**Review by alecthomas — APPROVED** [review]

File: google-java-format.hcl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+description = "Reformats Java source code to comply with Google Java Style.."
+runtime-dependencies = ["openjre-17.0.4.1_1"]
+binaries = ["google-java-format"]
+source = "https://github.com/google/google-java-format/releases/download/v{version}/google-java-format-{version}-all-deps.jar"
```
Comment (alecthomas) [inline google-java-format.hcl:4]: Variables need to be in the form `${version}` (note the `$`)

Comment (mxr) [reply google-java-format.hcl]: Oops. Fixed!

**Comment by alecthomas** [comment]
Just in case you're not aware, clang-format also supports Google's formatting for Java. Maybe it's not quite as correct though, not sure.

**Comment by mxr** [comment]
TIL, thanks

**Pull request merged**